src/sequence.lisp

```diff
@@ -196,31 +196,39 @@
 (defun copy-seq (sequence)
   (subseq sequence 0))
 
-
-;;; Reduce (based on SBCL's version)
-
-(defun reduce (function sequence &key key from-end (start 0) end (initial-value nil ivp))
+(defun %reduce (function sequence key index from-end step-fn compare-fn end value)
```

Comment:
I agree, I'm writing an iterative version of reduce.

Btw, how do you feel about including some of the ansi tests? Apparently in present of start end, the from end only should affect if the reduction is done left or right associatively.